File: Edward Poot/src/ql/ast/stmt/computedquestion.go
package stmt

import "ql/interfaces"

type ComputedQuestion struct {
	computation interfaces.Expr
	Question
}

func NewComputedQuestion(label interfaces.StringLiteral, varDecl interfaces.VarDecl, computation interfaces.Expr) ComputedQuestion {
	return ComputedQuestion{computation: computation, Question: NewQuestion(label, varDecl)}
}

func (this ComputedQuestion) Computation() interfaces.Expr {
	return this.computation
}

File: Mickeydus/QL/src/AST/form/ComputedQuestion.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package AST.form;

import AST.expressions.Expr;
import AST.types.*;
import typechecker.TypecheckInterface;

/**
 *
 * @author Dominique
 */
public class ComputedQuestion extends Question{
    
    private Expr computation;

    public ComputedQuestion(Ident ident, Label label, Type type) {
        super(ident, label, type);
    }
    
    public ComputedQuestion(Ident ident, Label label, Type type, Expr computation){
         super(ident, label, type);
        this.computation = computation;
    }
    
    public Expr getExpr()
    {
        return this.computation;
    }
    
//        public void addExprToList(Expr expression){
//        exprlist = new ArrayList();
//        exprlist.add(expression);
//    }
//    
//    public List<Expr> getExpressionList(){
//        return this.exprlist;
//    }

    @Override
    public <T> T accept(TypecheckInterface<T> visitor) {
      return visitor.visit(this);
    }
    
}